core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/block_time.go

```diff
@@ -0,0 +1,76 @@
+package logprovider
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+)
+
+var (
+	defaultSampleSize = int64(10)
+	// TODO: TBD
+	defaultBlockTime = int64(time.Second * 10)
+)
+
+type blockTimeResolver struct {
+	poller logpoller.LogPoller
+}
+
+func newBlockTimeResolver(poller logpoller.LogPoller) *blockTimeResolver {
+	return &blockTimeResolver{
+		poller: poller,
+	}
+}
+
+func (r *blockTimeResolver) BlockTime(ctx context.Context, blockSampleSize int64) (time.Duration, error) {
```

Comment:
this is not an ideal way to fetch the block time during startup instead of using a configuration. This block time is used to calculate recovery block (24 hr oldblock)